test/api/SettingTest.ts

```diff
@@ -211,5 +248,252 @@ describe('Setting tests', function() {
+      // Retrieve the crypto setting id
+      const read = await getCurrentCryptoData();
+      // Ensure current block size is 128
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        128, read.data.result[0].content.crypto.keyProperties.operationMode);
+      // Update block size to 192
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        192, read.data.result[0].content.crypto.keyProperties.operationMode);
+    });
+    it('Check crypto settings update - change key + block size (192->128)', async () => {
+      // Retrieve the crypto setting id
+      const read = await getCurrentCryptoData();
+      // Ensure current block size is 192
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        192, read.data.result[0].content.crypto.keyProperties.operationMode);
+      // Update block size to 128
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        128, read.data.result[0].content.crypto.keyProperties.operationMode);
+    });
+    it('Check crypto settings update - change key + block size (128->256)', async () => {
+      // Retrieve the crypto setting id
+      const read = await getCurrentCryptoData();
+      // Ensure current block size is 128
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        128, read.data.result[0].content.crypto.keyProperties.operationMode);
+      // Update block size to 256
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        256, read.data.result[0].content.crypto.keyProperties.operationMode);
+    });
+    it('Check crypto settings update - change key + operation mode (gcm->ctr)', async () => {
+      // Retrieve the crypto setting id
+      const read = await getCurrentCryptoData();
+      // Ensure current operation mode is gcm
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        read.data.result[0].content.crypto.keyProperties.blockSize, 'gcm');
+      // Update operation mode to ctr
+      await updateCryptoSettings(read.data.result[0], read.data.result[0].content.crypto.keyProperties.blockCypher,
+        read.data.result[0].content.crypto.keyProperties.blockSize, 'ctr');
+    });
+    it('Check crypto settings update - change key + algorithm (aes->camellia)', async () => {
+      // Retrieve the crypto setting id
+      const read = await getCurrentCryptoData();
+      // Ensure current algorithm is aes + ensure operation mode works with both camellia & aes
+      await updateCryptoSettings(read.data.result[0], 'aes',
+        read.data.result[0].content.crypto.keyProperties.blockSize, 'ctr');
```

Comment:
I would prefer that you use our default mode 'gcm' for key + algo change unit tests, we get then some more coverage on our defaults: read.data.result[0].content.crypto.keyProperties.operationMode should give it.

Reply:
That is how I initially tried, but we do not support gcm operation mode for camellia algorithm, which is why I changed it for this test.

Reply:
Use idea algo as a target then in these tests.

Reply:
'idea' algo still doesn't work with 'gcm'.
![image](https://user-images.githubusercontent.com/78014067/110444698-00c93b80-80c6-11eb-8525-13e5822590ef.png)
'aria' would work best. 
There's also this one: 
![image](https://user-images.githubusercontent.com/78014067/110444921-41c15000-80c6-11eb-9ae4-c9a29e761eae.png)
But here i am running into a different kind of error... looks like if i send as block size 'aes128' it fails because it would expect the block size to be a number.